engines/cpu.c

```diff
@@ -63,14 +179,65 @@ static enum fio_q_status fio_cpuio_queue(struct thread_data *td,
 		return FIO_Q_BUSY;
 	}
 
-	usec_spin(co->cpucycle);
+	switch (co->cpumode) {
+		case FIO_CPU_NOOP:
+			usec_spin(co->cpucycle);
+			break;
+		case FIO_CPU_QSORT:
+			do_qsort(td);
+			break;
+	}
```

Comment:
Probably wants to include a default: statement with a log_err() of some sort (and error return). Just a bit of defensive programming.

Reply:
I didn't on purpose. As I do a switch() on an enum, GCC would complain if any value of the num isn't present here. If any add a new value in the enum, GCC will request the developer to add the proper case statement.
A default statement, would kill that behavior
Is it fine for you ?

Reply:
Instead of having to worry about if this is true for any gcc that we use (and with the flags set), I'd prefer to have it as part of the init check where you have one. The rest don't need it. And it's still possible that someone sets this to a non-enum value, compiler won't warn about that.